src/main/java/com/wavefront/sdk/common/logging/DelegatingLogger.java

```diff
@@ -0,0 +1,64 @@
+package com.wavefront.sdk.common.logging;
+
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * Base class for delegating loggers.
+ *
+ * @author vasily@wavefront.com
+ */
+public abstract class DelegatingLogger extends Logger {
+  protected final Logger delegate;
+
+  /**
+   * @param delegate     Delegate logger.
+   */
+  public DelegatingLogger(Logger delegate) {
+    super(delegate.getName(), null);
+    this.delegate = delegate;
+  }
+
+  /**
+   * @param level   log level.
+   * @param message string to write to log.
+   */
+  @Override
+  public abstract void log(Level level, String message);
+
+  /**
+   * @param logRecord log record to write to log.
+   */
+  @Override
+  public void log(LogRecord logRecord) {
+    logRecord.setLoggerName(delegate.getName());
+    inferCaller(logRecord);
+    delegate.log(logRecord);
+  }
+
+  /**
+   * This is a JDK8-specific implementation. TODO: switch to StackWalker after migrating to JDK9+
+   */
+  private void inferCaller(LogRecord logRecord) {
+    StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
```

Comment:
this is quite expensive as an operation just FYI

Reply:
a warning should be somewhere in this class

Reply:
added comments and updated javadoc, but left this behavior as is for now so that the log messages can contain the right source class/method.
e.g., `com.wavefront.sdk.direct.ingestion.WavefrontDirectIngestionClient run
WARNING:...` instead of `com.wavefront.sdk.common.logging.DelegatingLogger log
WARNING:...`